Automattic/wp-calypso
# Domains: Self-serve domain mapping transfers to other site

#### Changes proposed in this Pull Request

The time is now: self-serve domain mappings 🎉 Splitting the work - this patch adds the ability to move domain mapping to a different site (it's simply easier to adjust this case than the user one).

#### Testing instructions

Requires D51163-code, but it's already been merged 🎉 

Go to Domain Management where you have a mapping you want to transfer. You should see a new action for it - Transfer mapping.
You need to be an admin on at least one other site to be able to transfer the mapping anywhere, of course.
And you need to be the owner of the mapping subscription - so be sure to check that case as well (probably need to adjust the copy there).

Step by step via screenshots:

<img width="751" alt="Screenshot 2020-10-21 at 14 10 11" src="https://user-images.githubusercontent.com/3392497/96732006-69c64600-13a7-11eb-9ed6-719c43dac8e2.png">
<img width="763" alt="Screenshot 2020-10-21 at 14 10 19" src="https://user-images.githubusercontent.com/3392497/96732044-72b71780-13a7-11eb-8383-f2160aaea0e8.png">
<img width="753" alt="Screenshot 2020-10-21 at 14 11 02" src="https://user-images.githubusercontent.com/3392497/96732056-764a9e80-13a7-11eb-9b04-ec9b087b0b0e.png">

For a PWPO user where the target site is not on a paid plan:
<img width="1311" alt="Screenshot 2020-10-28 at 19 45 44" src="https://user-images.githubusercontent.com/3392497/97488586-58a4a880-1956-11eb-8db4-4f832de66e1e.png">


PWPO-exempt user where the target site is not a paid plan:
<img width="1315" alt="Screenshot 2020-10-28 at 19 47 36" src="https://user-images.githubusercontent.com/3392497/97488721-7e31b200-1956-11eb-9182-1a0a68582f55.png">

When the target site is on a paid plan:
<img width="575" alt="Screenshot 2020-10-21 at 14 11 49" src="https://user-images.githubusercontent.com/3392497/96732058-76e33500-13a7-11eb-8216-0faccfba2f36.png">

Make sure you check these views also with an empty cache by adding `?flags=force-sympathy` to the URL.

Because of the special nature of VIP mappings, they are not supported in this initial version.
